protocol/interoperable-ether.md

```diff
@@ -0,0 +1,198 @@
+    if (messenger.crossDomainMessageSender() != address(this)) revert Unauthorized();
+
+    liquidity.source(amount);
+
+    bool success = SafeCall.transfer({ _target: to, _amount: amount });
+    require(success);
+  }
+}
+
+// Placed in genesis with a balance equal to type(uint248).max
+contract ETHLiquidity {
+  address weth internal = Predeploys.SUPERCHAIN_WETH;
+  L1Block internal l1Block = L1Block(Predeploys.L1_BLOCK);
+
+  function burn() external payable {
+    if (msg.sender != weth) revert Unauthorized();
+  }
+
+  function source(uint256 amount) {
+    if (msg.sender != weth) revert Unauthorized();
+    if (l1Block.isCustomGasToken()) revert OnlyEther();
+    require(SafeCall.transfer({ _target: weth, _value: amount }));
+  }
+}
+```
+
+### Longer term nice to haves
+
+- Enable ETH interface on `StandardBridge` for custom gas token chains and have it mint `SuperchainWETH` for deposits
+    - We can call this out of scope for now, but a future upgrade can make `SuperchainWETH` an `OptimismMintableERC20` to enable this
+
+# Open Questions
+
+## Naming
+
+We need to come up with a new name for this "superchain wrapped ether" to differentiate it. Not sure
+if it should be called `SWETH` or just stick with `WETH`. Without a different name, it will be confusing,
+but then it will create more overhead to get people to understand that its portable `WETH`.
+
+Since the usage of custom gas token is legible from within both L1 and L2, this means that the superchain `WETH`
+can block deposits of native asset when its a custom gas token chain.
+
+## `IOptimismMintableERC20` Support
+
+It is also possible to add in support for `IOptimismMintableERC20` so that `WETH` can be deposited directly into
+this predeploy. This would solve the problem of having `ETH` and native asset liquidity on the L2, since the chain
+operator likely needs to sell the earned native asset into `ether` to pay for DA. This could look like the `L1StandardBridge`
+automatically converting `ether` into `WETH` and then depositing it such that it mints the superchain `WETH` on L2.
+Without this solution, it means there is no easy way to get fungible `ether` on to a custom gas token chain.
+
+# Alternatives Considered
+
+## Upgrade Existing WETH Predeploy
+
+The existing `WETH` predeploy is not proxied, meaning the only way to upgrade
+it is with an irregular state transition. This is not a solution that we should
+take often, it adds technical debt that must be implemented in every execution
+layer client that supports OP Stack.
+
+Following this decision would be inconsistent with previous decision making where
+we decided specifically to remove native ether sends from within the protocol
+so that custom gas token chains can be interoperable.
+
+# Risks & Uncertainties
+
+- This adds a lot of bridge risk as its a new way to send `ether` between chains. We need to be sure to think in terms of state machines and invariants.
```

Comment:
Other risks:
- Placing `type(uint248).max` into ETHLiquidity, since we might introduce a bug that lets a disproportionate amount of ETH be withdrawn
- User confusion: Lots of new concepts, naming questions, etc.
- Are there any risks or footguns we introduce here by splitting up WETH supply between two predeploys? Want to think on the consequences there more

Reply:
These are great to add to the failure mode analysis